grails-app/services/org/broadinstitute/orsp/QueryService.groovy

```diff
@@ -567,19 +580,29 @@ class QueryService implements Status {
         issue
     }
 
+    Boolean areLinksApproved(String projectKey, String consentKey) {
+        List<ConsentCollectionLink> links = ConsentCollectionLink.findAll("from ConsentCollectionLink as ccl where ccl.projectKey= :projectKey and ccl.consentKey = :consentKey",
+                [consentKey: consentKey, projectKey: projectKey])
+        ArrayList approvedLinks = links.findAll {
```

Comment:
In general, favor List over ArrayList. Also, please use strong typing, i.e. `List<ConsentCollectionLink>`. I have a feeling that once you write a more concise query to get all approved statuses from the ccl table, you won't need to do this at all.